src/WinSparkleDotNet/WinSparkle.cs

```diff
@@ -0,0 +1,557 @@
+        ///     <para>
+        ///         Sets application build version number.
+        ///     </para>
+        ///     <para>
+        ///         This is the internal version number that is not normally shown to the user.
+        ///         It can be used for finer granularity that official release versions, e.g. for
+        ///         interim builds.
+        ///     </para>
+        ///     <para>
+        ///         If this function is called, then the provided *build* number is used for comparing
+        ///         versions; it is compared to the "version" attribute in the appcast and corresponds
+        ///         to OS X Sparkle's CFBundleVersion handling. If used, then the appcast must
+        ///         also contain the "shortVersionString" attribute with human-readable display
+        ///         version string. The version passed to win_sparkle_set_app_details()
+        ///         corresponds to this and is used for display.
+        ///     </para>
+        /// </summary>
+        ///
+        /// <remarks>
+        ///     <para>
+        ///         Can only be called @em before the first call to <see cref="Initialize"/>
+        ///     </para>
+        /// </remarks>
+        ///
+        /// <param name="buildVersion">   The version number. </param>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_set_app_build_version", CharSet = CharSet.Unicode, CallingConvention = CallingConvention.Cdecl)]
+        public static extern void SetAppBuildVersion(
+            [param: MarshalAs(UnmanagedType.LPWStr)] string buildVersion
+            );
+
+        /// <summary>   
+        ///     <para>
+        ///         Set the registry path where settings will be stored.
+        ///     </para>
+        ///     <para>
+        ///         Normally, these are stored in "HKCU\Software\&lt;company_name>\&lt;app_name>\WinSparkle"
+        ///         but if your application needs to store the data elsewhere for
+        ///         some reason, using this function is an alternative.
+        ///     </para>
+        ///     <para>
+        ///         Note that @a path is relative to HKCU/HKLM root and the root is not part of it.
+        ///     </para>
+        /// </summary>
+        /// 
+        /// <example>
+        ///     <code>SetRegistryPath(@"Software\My App\Updates");</code>
+        /// </example>
+        ///
+        /// <remarks>
+        ///     <para>
+        ///         Can only be called @em before the first call to <see cref="Initialize"/>
+        ///     </para>
+        /// </remarks>
+        ///
+        /// <param name="path">   Registry path where settings will be stored. </param>
+        [DllImport("WinSparkle.dll", EntryPoint = "win_sparkle_set_registry_path", CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
+        public static extern void SetRegistryPath(
+            [param: MarshalAs(UnmanagedType.LPWStr)] string path
+            );
```

Comment:
Why do you use `UnmanagedType.LPWStr` for that API function?
```
WIN_SPARKLE_API void __cdecl win_sparkle_set_registry_path(const char *path)
```
`UnmanagedType.LPWStr` would be a fit for a pointer to a null-terminated array of _Unicode_ characters. But then you explicitly selected `CharSet.Ansi`. Again, `UnmanagedType.LPStr` would be a better fit.